### PR TITLE
Bug 1522348 - Bulk assign open bugs to task, enhancement, defect field

### PR DESCRIPTION
**Pull request by kyoshino**
Happy SQL time! Add a script to do the migration task 1, 2, 3 written in [Bug 1524738](https://bugzilla.mozilla.org/show_bug.cgi?id=1524738#c6). Would like to test this on the dev server.

## Bugzilla link

[Bug 1522348 - Bulk assign open bugs to task, enhancement, defect field](https://bugzilla.mozilla.org/show_bug.cgi?id=1522348)

### CODE REVIEW

**Comment by dylanwh** [comment]
I'll run this against my database dump as soon as I get it

File: scripts/migrate-bug-type.pl [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+# defined by the Mozilla Public License, v. 2.0.
+
+use strict;
+use warnings;
+use lib qw(. lib local/lib/perl5);
+
+use Bugzilla;
+
+# List of products and components that use a bug type other than "defect"
+use constant MIGRATION_MAP => (
```
Comment (dylanwh) [inline scripts/migrate-bug-type.pl]:
```suggestion
my @MIGRATION_MAP => (
```

Comment (kyoshino) [reply scripts/migrate-bug-type.pl]: Thanks 😸

**Comment by kyoshino** [comment]
I'm moving this script to `/extensions/BMO/bin/` since this is BMO-specific.

**Comment by kyoshino** [comment]
Hmm this time the sanity test warns an untouched script 🤔 
```
#   Failed test 'extensions/BMO/bin/migrate-github-pull-requests.pl DOES NOT require Perl 5.10.1 --WARNING'
#   at t/002goodperl.t line 109.

#   Failed test 'extensions/BMO/bin/migrate-bug-type.pl DOES NOT require Perl 5.10.1 --WARNING'
#   at t/002goodperl.t line 109.
```

**Comment by kyoshino** [comment]
@dylanwh do you have any idea why these tests are failing. I’ve just moved the file under `extensions`.

**Comment by dklawren** [comment]
> 
> 
> @dylanwh do you have any idea why these tests are failing. I’ve just moved the file under `extensions`.

Since the file is not explicitly using Mojolicioius (which automatically use 5.10.1), you need to add:

use 5.10.1;

to the top of the file.

**Comment by kyoshino** [comment]
@dklawren Thanks! Somehow I was removing `use 5.10.1` from `migrate-github-pull-requests.pl` which made me confused myself. Added `use 5.10.1` to `migrate-bug-type.pl` so this should be fine now.

Speaking of the unexpected results, I’ve added a `if (scalar @$bug_ids)` check in case no bugs found in any listed component. Tested locally with a limited set of bugs/products/components and it worked. @dylanwh: can you please re-run this on the staging server?

**Comment by kyoshino** [comment]
Test done on the staging server.

**Comment by kyoshino** [comment]
We gotta run it later this week.

**Comment by kyoshino** [comment]
I’m checking if any of the recently added or removed components should be listed here to use the enhancement or task type. Cc @emceeaich

**Comment by kyoshino** [comment]
Good to go.

File: scripts/migrate-bug-type.pl [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  ['Websites', 'Web Analytics', 'task'],
+  ['Firefox Build System', 'Mach Core', 'enhancement'],
+  ['support.mozilla.org - Lithium', 'Feature request', 'enhancement'],
+);
+
+my $dbh = Bugzilla->dbh;
+
+$dbh->bz_start_transaction;
+
+# Disable the "enhancement" severity
```
[the remark this answers is not in the document]
Comment (kyoshino) [reply scripts/migrate-bug-type.pl]: Thanks, but somehow this says "outdated suggestions cannot be applied" so I'll make the change manually now.

**Pull request merged**